lib/Templating/DelegatingEngine.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Enterprise License (PEL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ * @license    http://www.pimcore.org/license     GPLv3 and PEL
+ */
+
+namespace Pimcore\Templating;

```

Comment:
Shouldn't we just use Symfony's `DelegatingEngine` and configure it properly, e.g.: 

```yml
    pimcore.templating.engine.delegating:
        public: true
        class: Symfony\Component\Templating\DelegatingEngine
        arguments: ['@twig']
```

?

Reply:
Yes, that was my first thought. Unfortunately, it does not support `Twig\Environment` directly so we need to introduce a wrapper. Also, overriding DelegatingEngine here means we can switch the case for Twig or custom engines(TwigEngine: Twig Wrapper, PhpEngine), so it is still faster for instances using just Twig.